packages/web/components/place-limit-tool/index.tsx

```diff
@@ -0,0 +1,652 @@
+import { Dec, IntPretty, PricePretty } from "@keplr-wallet/unit";
+import { DEFAULT_VS_CURRENCY } from "@osmosis-labs/server";
+import { isValidNumericalRawInput } from "@osmosis-labs/utils";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import { parseAsString, parseAsStringLiteral, useQueryStates } from "nuqs";
+import {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+
+import { Icon } from "~/components/assets/icon";
+import {
+  AssetFieldset,
+  AssetFieldsetFooter,
+  AssetFieldsetHeader,
+  AssetFieldsetHeaderBalance,
+  AssetFieldsetHeaderLabel,
+  AssetFieldsetInput,
+  AssetFieldsetTokenSelector,
+} from "~/components/complex/asset-fieldset";
+import { LimitPriceSelector } from "~/components/place-limit-tool/limit-price-selector";
+import { TRADE_TYPES } from "~/components/swap-tool/order-type-selector";
+import { PriceSelector } from "~/components/swap-tool/price-selector";
+import { TradeDetails } from "~/components/swap-tool/trade-details";
+import { Button } from "~/components/ui/button";
+import { EventPage } from "~/config";
+import {
+  useDisclosure,
+  useSlippageConfig,
+  useTranslation,
+  useWalletSelect,
+} from "~/hooks";
+import { usePlaceLimit } from "~/hooks/limit-orders";
+import { AddFundsModal } from "~/modals/add-funds";
+import { ReviewOrder } from "~/modals/review-order";
+import { useStore } from "~/stores";
+import { formatPretty } from "~/utils/formatter";
+import { countDecimals, trimPlaceholderZeros } from "~/utils/number";
+
+export interface PlaceLimitToolProps {
+  page: EventPage;
+  refetchOrders: () => Promise<any>;
+}
+
+const fixDecimalCount = (value: string, decimalCount = 18) => {
+  const split = value.split(".");
+  const result =
+    split[0] +
+    (decimalCount > 0 ? "." + split[1].substring(0, decimalCount) : "");
+  return result;
+};
+
+const transformAmount = (value: string, decimalCount = 18) => {
+  let updatedValue = value;
+  if (value.endsWith(".") && value.length === 1) {
+    updatedValue = value + "0";
+  }
+
+  if (value.startsWith(".")) {
+    updatedValue = "0" + value;
+  }
+
+  const decimals = countDecimals(updatedValue);
+  return decimals > decimalCount
+    ? fixDecimalCount(updatedValue, decimalCount)
+    : updatedValue;
+};
```

Comment:
This code is also used on [limit-input](https://github.com/osmosis-labs/osmosis-frontend/pull/3683/files#diff-1f48e9195e251a059b987e36d586c93855cf081f1380cd73ccce0f8f35ff4e54).  I suggest creating a file to share this util and write some simple unit tests.

Reply:
Limit input was moved to a separate component. I've removed it completely.